integrations/sonarqube/main.py

```diff
@@ -18,64 +25,94 @@ def init_sonar_client() -> SonarQubeClient:
     )
 
 
+def produce_component_params(
+    client: SonarQubeClient, selector: Any, initial_params: dict[str, Any] = {}
+) -> dict[str, Any]:
+    component_query_params: dict[str, Any] = {}
+    if client.organization_id:
+        component_query_params["organization"] = client.organization_id
+
+    ## Handle query_params based on environment
+    if client.is_onpremise:
+        if initial_params:
+            component_query_params.update(initial_params)
+        elif event.resource_config:
+            # This might be called from places where event.resource_config is not set
+            # like on_start() when creating webhooks
+
+            selector = cast(CustomSelector, event.resource_config.selector)
+            component_query_params.update(selector.generate_request_params())
+    return component_query_params
+
+
 sonar_client = init_sonar_client()
 
 
 @ocean.on_resync(ObjectKind.PROJECTS)
 async def on_project_resync(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
-    logger.info(f"Listing Sonarqube resource: {kind}")
-    fetched_projects = False
-    async for project_list in sonar_client.get_all_projects():
-        yield project_list
-        fetched_projects = True
-
-    if not fetched_projects:
-        logger.error("No projects found in Sonarqube")
-        raise RuntimeError(
-            "No projects found in Sonarqube, failing the resync to avoid data loss"
-        )
```

Comment:
does it mean that if a customer is still using this kind data can be deleted if no data is returned from the API?

Reply:
Hmm yes, that's true. Do you want me to restore the behaviour where we prevent data from being deleted.